src/Helpers/DataResolver.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Firesphere\SolrSearch\Helpers;
+
+use Exception;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBField;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\View\ArrayData;
+
+class DataResolver
+{
+    protected $component;
+    protected $columns = [];
+    protected $columnName = '';
+
+    public function __construct($component, $columns = [])
+    {
+        if (!is_array($columns)) {
+            $columns = array_filter(explode('.', $columns));
+        }
+        $this->columns = $columns;
+        $this->component = $component;
+        $this->columnName = $this->columns ? array_shift($this->columns) : null;
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $obj
+     * @param array|string $columns
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    public static function identify($obj, $columns = [])
+    {
+        $instance = new self($obj, $columns);
```

Comment:
Maybe not instantiate a `new` all the time, but use Injector and get/set methods?

Reply:
That is a really good point :+1: I always thought that this is much faster than a factory class since the new object is already known by the class itself. 
What I will do is compare how fast this will be when I replace it with Injector vs classic way in running PHPUnit tests.